examples/fluids/src/setupts.c

```diff
@@ -450,6 +450,7 @@ PetscErrorCode TSSolve_NS(DM dm, User user, AppCtx app_ctx, Physics phys, Vec *Q
   }
   PetscCall(TSSetMaxTime(*ts, 500. * user->units->second));
   PetscCall(TSSetExactFinalTime(*ts, TS_EXACTFINALTIME_STEPOVER));
+  PetscCall(TSSetErrorIfStepFails(*ts, PETSC_FALSE));
```

Comment:
Does this also remove the error output if TS exits with a failure? ie:

```
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR: Invalid argument
[0]PETSC ERROR: Wrong type of object: Parameter # 1
[0]PETSC ERROR: WARNING! There are option(s) set that were not used! Could be the program crashed before they were used or a spelling mistake, etc!
[0]PETSC ERROR: Option left: name:-spanstats_dm_sparse_localize value: 0 source: code
   [......]
[0]PETSC ERROR: Petsc Development GIT revision: v3.18.3-328-gecdab737cc0  GIT Date: 2023-01-02 11:45:34 -0700
[0]PETSC ERROR: ./navierstokes on a arch-dev named fermat by jrwrigh Fri Jan  6 20:21:15 2023
[0]PETSC ERROR: Configure options --with-64-bit-indices=0 --download-hdf5 --download-cgns --download-tetgen=1 --download-yaml=1 PETSC_ARCH=arch-dev
[0]PETSC ERROR: #1 PetscFEGetHeightSubspace() at /home/jrwrigh/software/petsc/src/dm/dt/fe/interface/fe.c:1692
   [......]
[0]PETSC ERROR: PETSc Option Table entries:
[0]PETSC ERROR: -bc_inflow 6 (source: file)
```

In the case of a failure, it's probably not bad to still include this information just in case it can help illuminate something (mispelled options, wrong Petsc version, etc.)

Reply:
:point_up:  That error hasn't even gotten to TS.

It only prevents TS divergence (perhaps because SNES and KSP diverged, or because an adaptive controller gives up) from raising an error.

Reply:
Right, I just posted that error message as an example of useful info that's put into the error message. Particularly, the `option(s) set that were not used` part is particularly useful.

It's not uncommon to run into simple mistakes (such as misspelled options) when bootstrapping a new simulation for the first time and having those breadcrumbs to verify what the code is doing can be quite helpful.

Reply:
Good point. Arguably we should always use `-options_left` (or our code can print the equivalent). It's part of `-log_view`, which is typically always used for provenance in production runs, but noisy for routine testing.